kotlin-inject-runtime/src/nativeMain/kotlin/me/tatarka/inject/internal/LazyMap.kt

```diff
@@ -1,13 +1,13 @@
 package me.tatarka.inject.internal
 
-import kotlinx.atomicfu.locks.ReentrantLock
+import kotlinx.atomicfu.locks.SynchronizedObject
 import kotlinx.atomicfu.locks.withLock
 
 private val NULL = Any()
 
 actual class LazyMap {
     private val map = mutableMapOf<String, Any>()
-    private val lock = ReentrantLock()
+    private val lock = SynchronizedObject()
```

Comment:
https://github.com/Kotlin/kotlinx-atomicfu/issues/401

Reply:
is `SynchronizedObject` reentrant? Since we call into client code inside the lock I think it's very possible. Don't think we have a good test for this unfortunately.

Reply:
Looks like yes:
https://github.com/Kotlin/kotlinx-atomicfu/commit/3b895fe0409d5195877d813581edeb88cf014774#diff-3a82b7dd4d92bba2b14c26b50c02172b206613984ac478fc4d29b432f508f88cR29

https://docs.oracle.com/javase/tutorial/essential/concurrency/locksync.html

Reply:
I believe `ReentrantLock` is just a typealias for `SynchronizedObject` on Native.